scripts/main.js

```diff
@@ -0,0 +1,57 @@
+const fetchResult = async (n) => {
+  try {
+    const response = await fetch(`${baseUrl}${n}`);
+    const contentType = response.headers.get('content-type');
+    if (contentType.includes('application/json')) {
+      const data = await response.json();
+      return data.result;
+    } else {
+      const data = await response.text();
+      switchClass("#result", "red", true);
+      dontPanic();
+      return `Server Error: ${data}`
```

Comment:
not sure what you mean here, could you please explain?

Reply:
After you get the data and convert it to text, you can throw an error with the data, so in line 12 you can write:
`throw new Error(data) `
instead of returning. 

After the error is thrown, the catch will catch it. So it will be something like this: 

`catch(err) {
       err will be the error that was thrown in the throw new Error (the data)
    /// Append the err to the HTML
  }`